#VRML_SIM R2019a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.5779521105411721 0.7898424185329204 0.20523233616336317 0.8142916899525948
  position 2.319364915199812 1.3879885646781174 2.187195250819425
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
}
Robot {
  translation 0.1 0.07999999999999997 0.1
  children [
    DEF GR_BODY Group {
      children [
        DEF ROBOT_BODY Shape {
          appearance PBRAppearance {
            baseColor 0.9372549019607843 0.1607843137254902 0.1607843137254902
          }
          geometry DEF G_BODY Box {
            size 0.25 0.07 0.1
          }
        }
        Transform {
          translation 0.11 -0.05 0
          children [
            Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Sphere {
                radius 0.015
              }
            }
          ]
        }
      ]
    }
    DEF WHEEL1_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor -0.05 0 0.06
      }
      device [
        PositionSensor {
          name "encoder1"
        }
        RotationalMotor {
          name "motor_right"
          maxVelocity 52.3598
          maxTorque 0.5931
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation -0.04986351996991146 0.002512121656639714 0.060000000000000005
        rotation 1 0 0 1.57
        children [
          DEF SH_WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry DEF G_WHEEL Cylinder {
              height 0.02
              radius 0.075
            }
          }
        ]
        boundingObject USE G_WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
        anchor -0.05 0 -0.06
      }
      device [
        PositionSensor {
          name "encoder2"
        }
        RotationalMotor {
          name "motor_left"
          maxVelocity 52.3598
          maxTorque 0.5931
        }
      ]
      endPoint Solid {
        translation -0.05161149370379927 0.0010565499530129716 -0.06
        rotation 1 0 0 1.57
        children [
          DEF WH_SHAPE2 Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.075
            }
          }
        ]
        name "solid(1)"
        boundingObject USE WH_SHAPE2
        physics Physics {
        }
      }
    }
    DistanceSensor {
      translation 0.125 0 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.45098039215686275 0.8235294117647058 0.08627450980392157
          }
          geometry Box {
            size 0.02 0.01 0.01
          }
        }
      ]
      name "distance_sensor"
      lookupTable [
        0 0 0
        0.02 0 0
        5 1023 0
      ]
      aperture 0.5235
      resolution 1
    }
  ]
  boundingObject USE GR_BODY
  physics Physics {
    mass 3
  }
  controller "u2_exam_Zamudio_Shadeen"
}
